File: src/main/java/com/codecool/cashtrack/domain/entities/ids/FriendRequestId.java
package com.codecool.cashtrack.domain.entities.ids;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.util.Objects;

@Getter
@Setter
@Embeddable
public class FriendRequestId implements Serializable {
    @Column(name = "from_user_id", nullable = false)
    private Long from; // user id

    @Column(name = "to_user_id", nullable = false)
    private Long to;   // user id

    public FriendRequestId(Long from, Long to) {
        this.from = from;
        this.to = to;
    }

    public FriendRequestId() {

    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        if (!(o instanceof FriendRequestId that)) return false;
        return from.equals(that.from) && to.equals(that.to);
    }

    @Override
    public int hashCode() {
        return Objects.hash(from, to);
    }
}

File: src/main/java/com/codecool/repvault/domain/entities/ids/FriendRequestId.kt
package com.codecool.repvault.domain.entities.ids

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import java.io.Serializable
import java.util.*


@Embeddable
class FriendRequestId : Serializable {
    @Column(name = "from_user_id", nullable = false)
    var from: Long? = null // user id

    @Column(name = "to_user_id", nullable = false)
    var to: Long? = null // user id

    constructor(from: Long, to: Long) {
        this.from = from
        this.to = to
    }

    constructor()

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null) return false
        if (o !is FriendRequestId) return false
        return from == o.from && to == o.to
    }

    override fun hashCode(): Int {
        return Objects.hash(from, to)
    }
}